csrc/dynamic_transform.cpp

```diff
@@ -270,8 +352,11 @@ void DynamicTransformConcretizer::mutate(TensorView* tv) {
   // axes inherited from the producers
   auto propagated = propagateFromProducerToConsumer(tv);
 
-  // If no root domain is altered, nothing to do further
+  // If no root domain is altered by producer, we don't need to propagate back
+  // up to rfactor, so do a simple mutation.
   if (!propagated) {
+    mutate(tv->domain());
+    OptOutMutator::mutate(tv);
```

Comment:
This looks trivial, but any future change would need to be applied to both here and at the end of this function, which could be a source of future bugs. I think we should just remove this if branch and leave a note that even if `!propagated`, there can be things, like concretization of symbolic domains due to dynamic resize ops.

Reply:
Good point. Reducing the complexity here probably outweighs whatever gains we might get from returning early here or `continue`ing in the loop.